addons/ophthalmology/functions/fnc_treatmentAdvanced_eyeShield.sqf

```diff
@@ -32,40 +41,23 @@ if ((_eyeInjuries find 0) == 0) then {
         _this params ["_args", "_pfhID"];
         _args params ["_unit", "_activeEye"];
     
-        if ((hmd _unit) != "kat_eyecovers_left") exitWith {
+        if ((hmd _unit) != _shieldItem) exitWith {
             _pfhID call CBA_fnc_removePerFrameHandler;
             "KAT_EyeShield" cutText ["","PLAIN",0,true];
         };
     
-        private _eyeInjury = _unit getVariable ["kat_ophthalmology_eyeInjuries", [1, 1]];
-        _unit setVariable ["kat_ophthalmology_eyeInjuries", [(((_eyeInjury select 0) + 0.001) min 1), (_eyeInjury select 1)], true];
+        private _eyeInjury = _unit getVariable [QGVAR(eyeInjuries), [1, 1]];
+        _unit setVariable [QGVAR(eyeInjuries), [(((_eyeInjury select 0) + 0.001) min 1), (_eyeInjury select 1)], true];
     }, 30, [
         _patient,
         _activeEye
     ]] call CBA_fnc_addPerFrameHandler;
+};
 
+if ((_eyeInjuries find 0) == 0) then {
```

Comment:
If checking for a specific element in a specific index, use select so the entirety of _eyeInuries isn't iterated on.

`(_eyeInjuries select 0) == 0`